classwork/classwork5.1/MegaDataPool.cpp

```diff
@@ -0,0 +1,50 @@
+#include "MegaData.cpp"
+
+class MegaDataPool {
+public:
+    MegaDataPool(size_t poolSize) : poolSize(poolSize), usedCount(0) {
+        pool.reserve(poolSize);
+        for (size_t i = 0; i < poolSize; ++i) {
+            pool.emplace_back();
+        }
+        availableIndices.reserve(poolSize);
+        for (size_t i = 0; i < poolSize; ++i) {
+            availableIndices.push_back(i);
+        }
+    }
+
+    MegaData* acquire() {
```

Comment:
in general usage of & is safer then * as it prevents accidental delete call. You can just return back & to release call and rely of memory address of that object e.g. release(/*MegaData&*/ obj){ MegaData* ptr = &obj;}

Reply:
and of course it should be [[nodiscard]] as accidental ignore of returned value will cause memory-leak-like behaviour as no one will release that object